### PR TITLE
feat: switch to GitHub workflow to `terraform apply`

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Switching to a GitHub `terraform apply` workflow to reduce the number of third
party tools that have access to our secrets..

### CODE REVIEW

File: .github/workflows/terraform-apply.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  terraform-apply:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@3d8debd658c92063839bc97da5c2427100420dec # v1.3.2
```
Comment (mohamed-cds) [inline .github/workflows/terraform-apply.yml:27]: could we use `cds-snc/terraform-tools-setup@v1` instead ?

Comment (patheard) [reply .github/workflows/terraform-apply.yml:27]: Good idea - I'll switch

**Review by CalvinRodo — APPROVED** [review]
:taco::cat:

**Review by mohamed-cds — APPROVED** [review]
🚀 ✅

**Pull request merged**